server/remote_cache/digest/digest.go

```diff
@@ -343,6 +343,40 @@ func ElementsMatch(s1 []*repb.Digest, s2 []*repb.Digest) bool {
 	return true
 }
 
+// Diff returns the differences between two slices of digests. If the slices differ in the count of a non-unique element,
+// that does not count as a difference
+//
+// missingFromS1 contains the digests that are in S2 but not S1
+// missingFromS2 contains the digests that are in S1 but not S2
+func Diff(s1 []*repb.Digest, s2 []*repb.Digest) (missingFromS1 []*repb.Digest, missingFromS2 []*repb.Digest) {
+	missingFromS1 = make([]*repb.Digest, 0)
+	missingFromS2 = make([]*repb.Digest, 0)
+
+	s1Set := make(map[*repb.Digest]struct{}, len(s1))
+	for _, d := range s1 {
+		s1Set[d] = struct{}{}
+	}
+
+	s2Set := make(map[*repb.Digest]struct{}, len(s2))
+	for _, d := range s2 {
+		s2Set[d] = struct{}{}
+
+		_, inS1 := s1Set[d]
+		if !inS1 {
```

Comment:
nit: in go you can inline the variable definition into the if clause if it's not needed outside the if statement (i.e. `if _, inS1 := s1Set[d]; !inS1 {`

Reply:
Done